card/lib/card/model/save_helper.rb

```diff
@@ -10,11 +10,9 @@ module Model
     # to resolve name conflicts)
     module SaveHelper
       def as_user user_name
-        current = Card::Auth.current_id
-        Card::Auth.current_id = Card.fetch_id user_name
-        yield
-      ensure
-        Card::Auth.current_id = current
+        Card::Auth.with current_id: Card.fetch_id(user_name) do
+          yield
+        end
```

Comment:
I know this is old, but this isn't usually the way we use "as", right?  Doesn't "as" change the permissions but not the current user?  (Eg, if I'm PK and I do something "as_bot", I get bot permissions but the action is still performed by PK.